src/encodings.jl

```diff
@@ -1,271 +1,384 @@
+    "ISO88595", "ISO88596", "ISO88597", "ISO88598", "ISO88599", "ISO88599E",
+    "windows-1250", "WINDOWS-1250", "windows-1251", "WINDOWS-1251", "windows-1252",
+    "WINDOWS-1252", "windows-1253", "WINDOWS-1253", "windows-1254", "WINDOWS-1254",
+    "windows-1255", "WINDOWS-1255", "windows-1256", "WINDOWS-1256",
+    "windows-1257", "WINDOWS-1257", "windows-1258", "WINDOWS-1258",
+    "WINDOWS-31J", "WINDOWS-50221", "WINDOWS-51932", "windows-874",
+    "WINDOWS-874", "WINDOWS-932", "WINDOWS-936"]
+
+# 16-bit codeunit encodings
+const encodings16 = [
+    "UTF-16", "UTF-16BE", "UTF-16LE"
+    ]
+
+# 32-bit codeunit encodings
+const encodings32 = [
+    "UTF-32", "UTF-32BE", "UTF-32LE", "UTF32", "UTF32BE", "UTF32LE"
+    ]
+
+const encodings_other = [
+    "1026", "1046", "1047", "10646-1:1993", "10646-1:1993/UCS4",
+    "437", "500", "500V1", "850", "851", "852", "855", "856", "857",
+    "860", "861", "862", "863", "864", "865", "866", "866NAV", "869",
+    "874", "8859_1", "8859_2", "8859_3", "8859_4", "8859_5", "8859_6",
+    "8859_7", "8859_8", "8859_9", "904", "ANSI_X3.110", "ANSI_X3.110-1983",
+    "ANSI_X3.4", "ANSI_X3.4-1968", "ANSI_X3.4-1986", "ARABIC", "ARABIC7",
+    "ARMSCII-8", "ASCII", "ASMO_449", "ASMO-708", "BALTIC", "BIG-5",
+    "BIG-FIVE", "big5", "BIG5", "big5-hkscs", "BIG5-HKSCS", "BIG5-HKSCS:1999",
+    "BIG5-HKSCS:2001", "BIG5-HKSCS:2004", "BIG5-HKSCS:2008", "big5hkscs",
+    "BIG5HKSCS", "BIGFIVE", "BRF", "BS_4730", "C99", "CA", "CHINESE",
+    "CN", "CN-BIG5", "CN-GB", "CN-GB-ISOIR165", "CP-AR", "CP-GR",
+    "CP-HU", "CP-IS", "CPIBM861", "CSA_T500", "CSA_T500-1983", "CSA_Z243.4-1985-1",
+    "CSA_Z243.4-1985-2", "CSA_Z243.419851", "CSA_Z243.419852", "CSA7-1",
+    "CSA7-2", "CSBIG5", "CSDECMCS",
+    "CSEUCKR", "CSEUCPKDFMTJAPANESE", "CSEUCTW", "CSGB2312", "CSHALFWIDTHKATAKANA",
+    "CSHPROMAN8", "CSISO10367BOX", "CSISO103T618BIT", "CSISO10SWEDISH",
+    "CSISO111ECMACYRILLIC", "CSISO11SWEDISHFORNAMES", "CSISO121CANADIAN1",
+    "CSISO122CANADIAN2", "CSISO139CSN369103", "CSISO141JUSIB1002",
+    "CSISO143IECP271", "CSISO14JISC6220RO", "CSISO150", "CSISO150GREEKCCITT",
+    "CSISO151CUBA", "CSISO153GOST1976874", "CSISO159JISX02121990",
+    "CSISO15ITALIAN", "CSISO16PORTUGESE", "CSISO17SPANISH", "CSISO18GREEK7OLD",
+    "CSISO19LATINGREEK", "CSISO2022CN", "csISO2022JP", "CSISO2022JP",
+    "CSISO2022JP2", "CSISO2022KR", "CSISO2033", "CSISO21GERMAN",
+    "CSISO25FRENCH", "CSISO27LATINGREEK1", "CSISO49INIS", "CSISO4UNITEDKINGDOM",
+    "CSISO50INIS8", "CSISO51INISCYRILLIC", "CSISO5427CYRILLIC", "CSISO5427CYRILLIC1981",
+    "CSISO5428GREEK", "CSISO57GB1988", "CSISO58GB1988", "CSISO58GB231280",
+    "CSISO60DANISHNORWEGIAN", "CSISO60NORWEGIAN1", "CSISO61NORWEGIAN2",
+    "CSISO646DANISH", "CSISO69FRENCH", "CSISO84PORTUGUESE2", "CSISO85SPANISH2",
+    "CSISO86HUNGARIAN", "CSISO87JISX0208", "CSISO88GREEK7", "CSISO89ASMO449",
+    "CSISO90", "CSISO92JISC62991984B", "CSISO99NAPLPS", "CSISOLATIN1",
```

Comment:
Oh, yes, I was not advocating removing anything, simply meant that I think we should separate out handling of all of the aliases, and just have a table entry for each truly unique encoding.